src/components/SidebarNavigationItem/SidebarNavigationItem.tsx

```diff
@@ -1,84 +1,136 @@
-import { Icon } from "@/components";
-import { SidebarAccordion } from "@/components/Accordion/Accordion";
+import styled from "styled-components";
 import { IconName } from "@/components/Icon/types";
 
-import styled from "styled-components";
+import { Collapsible, FlexContainer, IconDir } from "./Collapsible";
+import { ComponentPropsWithoutRef, ElementType, HTMLAttributes, ReactNode } from "react";
+import { Icon } from "..";
 
-interface DefaultSidebarNavigationItemProps
-  extends React.HTMLAttributes<HTMLButtonElement> {
-  collapsible?: false | undefined | null;
-  label?: undefined;
-  icon?: IconName;
-  children: React.ReactNode;
+interface DefaultSidebarNavigationItemProps<T extends ElementType> {
+  label?: never;
+  component?: T;
```

Comment:
I added the component inorder to allow the a href or any other element 
We are currently using button

Reply:
that's the point, this only needs to be either a button or a link. Let's add a href prop. If it's specified it will behave as a link otherwise as a button.